### PR TITLE
Block: treeshaking tasks

### PR DESCRIPTION
**Pull request by ScottyPoi**
Block Treeshaking task cleanup from #3560

- [x] Kill off `from-rpc.ts`, `header-from-rpc.ts` files, move code within function blocks of `constructors.ts`
- [x] Create `block` and `header` directories, move `block.ts` and `header.ts` over, create `index.ts` file, split `constructors.ts` into the block and header methods and move over respectively (if this full TODO item is controversial we can also give a quick discussion, I would find it a useful better organizing structure)
- [x] Rename `createBlockFromValuesArray` -> `createBlockFromBytesArray`
- [x] Rename `createBlockHeaderFromValuesArray` -> `createBlockHeaderFromBytesArray`

### CODE REVIEW

File: packages/block/src/block/constructors.ts [hunk 1]
```diff
@@ -57,71 +59,25 @@ import type {
...
-/**
- * Static constructor to create a block header from a header data dictionary
- *
- * @param headerData
- * @param opts
- */
-export function createBlockHeader(headerData: HeaderData = {}, opts: BlockOptions = {}) {
-  return new BlockHeader(headerData, opts)
-}
+export function normalizeTxParams(_txParams: any) {
```
Comment (acolytec3) [inline packages/block/src/block/constructors.ts:62]: Two things on this function:
1) It isn't a constructor so ideally we put it in a helpers/utils file.  
2) We have essentially the same function [here](https://github.com/ethereumjs/ethereumjs-monorepo/blob/659d4280609943effe5322dcec7747a0cb6e50f8/packages/tx/src/fromRpc.ts#L5).  Can we harmonize these two and consolidate in one location (maybe in `@ethereumjs/util/helpers.ts` or `internal.ts`)  The only real difference I see is converting `0x0` to `0x` in the transaction signatuere values.

**Review by acolytec3 — DISMISSED** [review]
I generally really like this approach.  One nit about `normalizeTxParams` that we should address (though can be in a different PR if you want).

**Review by holgerd77 — APPROVED** [review]
LGTM

File: packages/block/src/block/constructors.ts [hunk 2]
```diff
@@ -10,9 +10,9 @@ import {
   CLRequestFactory,
   ConsolidationRequest,
```
Comment (Jamiecaty) [inline packages/block/src/block/constructors.ts:11]: - [ ]

**Review by Jamiecaty — CHANGES_REQUESTED** [review]

**Review by Jamiecaty — APPROVED** [review]

**Pull request merged**